api/src/main/java/jakarta/enterprise/lang/model/AnnotationTarget.java

```diff
@@ -7,37 +7,98 @@
 import java.util.function.Predicate;
 
 /**
- * Annotation target is anything that can be annotated.
+ * An Annotation target is anything that can be annotated.
  * That is:
  *
  * <ul>
- * <li>a <i>declaration</i>, such as a class, method, field, etc.</li>
- * <li>a <i>type parameter</i>, occuring in class declarations and method declarations</li>
- * <li>a <i>type use</i>, such as a type of method parameter, a type of field, a type argument, etc.</li>
+ * <li>a <i>declaration</i>, such as a class, method, field, etc. See {@link jakarta.enterprise.lang.model.declarations.DeclarationInfo}.</li>
+ * <li>a <i>type parameter</i>, occurring in class declarations and method declarations.</li>
+ * <li>a <i>type use</i>, such as a type of method parameter, a type of field, a type argument, etc. See {@link jakarta.enterprise.lang.model.types.Type}.</li>
  * </ul>
+ *
+ *
+ * @since 4.0.0
  */
 public interface AnnotationTarget {
     // TODO specify equals/hashCode (for the entire .lang.model hierarchy)
     // TODO settle on using Optional everywhere, or allowing null everywhere; it's a mix right now
+    // TODO what about declared vs not declared annotations?
+    // TODO I am wondering why we are deviating from javax.lang.model names here like AnnotatedConstruct, Element etc.
```

Comment:
To be honest, I find the `javax.lang.model` hierarchy quite counter-intuitive (e.g. using `VariableElement` for both fields and parameters). The Portable Extensions types are much nicer, and I'd be in favor of moving towards similar names -- except that makes for rather poor coding experience.